documentation/src/js_console.md

```diff
@@ -0,0 +1,116 @@
+
+# Forest JavaScript Console
+
+Forest console stands as an alternative to tools like [Curl](https://github.com/curl/curl) or other subcommands found in `forest-cli` for interacting with the Filecoin JSON-RPC API.
+
+## Starting the console
+
+`forest-cli attach` can be used to open a Javascript console connected to your Forest node.
+
+Like for some other `forest-cli` subcommands you will need to pass an admin `--token` given what endpoints you will call.
+
+For a description of different options please refer to the developer documentation CLI page.
+
+## Interactive Use
+
+First start Forest node inside another terminal:
+
+```bash
+./target/release/forest --chain calibnet --encrypt-keystore false
+```
+
+Copy the Admin token printed in the logs and set it to an environment variable, ie:
+
+```bash
+TOKEN="..."
+```
+
+To attach to your Forest node, run `forest-cli` with the `attach` subcommand:
+
+```bash
+./target/release/forest-cli --token $TOKEN attach 
+```
+
+You should now see a prompt and be able to interact:
+
+```                                          
+Welcome to the Forest Javascript console!
+
+To exit, press ctrl-d or type :quit
+> console.log("Forest running on " + chainGetName())
+Forest running on calibnet
+```
+
+You can directly call JSON-RPC API endpoints that have been binded to the console.
+
+For example `Filecoin.ChainGetName` will be binded to the global `chainGetName` function.
+
+### Tips
+
+- Console history is saved in your `~/.forest_history`
+- Use `:clear` to erase session commands before exiting
+- Use `_BOA_VERSION` to get engine version
+
+## Non-interactive Use
+
+### Exec Mode
+
+It is also possible to execute commands non-interactively by passing `--exec` flag and a JavaScript snippet to `forest-cli attach`. The result is displayed directly in the terminal rather than in the interactive console.
+
+For example, to display the current epoch:
+
+```bash
+./target/release/forest-cli attach --exec "syncStatus().ActiveSyncs[0].Epoch"
+```
+
+Or print node version:
+
+```bash
+./target/release/forest-cli attach --exec "commonVersion()"
```

Comment:
It'd be nice to be able to pass an entire JS file to `exec` or some other flag/subcommand

Reply:
Good idea but to keep this PR not to big, we can add address this in a future PR (maybe with some other enhancements, like command line completion etc).

In the meantime you can do:
```
$ ./forest-cli attach --exec "const _Foo = require('foo.js')"
hi
```
where foo.js:
```
console.log("hi");
```